src/mbgl/text/shaping.cpp

```diff
@@ -326,15 +326,36 @@ void shapeLines(Shaping& shaping,
                 const GlyphMap& glyphMap,
                 bool allowVerticalPlacement) {
     float x = 0;
-    float y = Shaping::yOffset;
-    
+    float y = 0;
     float maxLineLength = 0;
+    bool hasBaseline{false};
 
+    for (std::size_t i = 0; i < lines.size(); ++i) {
```

Comment:
If we would cache 'has baseline' for FontStack (FontMetrics), we would not need to do lookup for each glyph, and just go through each section / fontstack.